src/pyhf/tensor/pytorch_backend.py

```diff
@@ -327,8 +329,17 @@ def normal_cdf(self, x, mu=0.0, sigma=1.0):
         Returns:
             PyTorch FloatTensor: The CDF
         """
-        normal = torch.distributions.Normal(mu, sigma)
-        return normal.cdf(x)
+        # the implementation of torch.Normal.cdf uses torch.erf:
+        # 0.5 * (1 + torch.erf((value - self.loc) * self.scale.reciprocal() / math.sqrt(2)))
+        # (see https://github.com/pytorch/pytorch/blob/3bbedb34b9b316729a27e793d94488b574e1577a/torch/distributions/normal.py#L78-L81)
+        # we get a more numerically stable variant for low p-values/high significances using erfc(x) := 1 - erf(x)
+        # since erf(-x) = -erf(x) we can replace
+        # 1 + erf(x) = 1 - erf(-x) = 1 - (1 - erfc(-x)) = erfc(-x)
+        mu, sigma = [
+            torch.as_tensor(_par, dtype=self.dtypemap["float"])
```

Comment:
added this, since we can't rely on the `torch.set_default_dtype(self.dtypemap["float"])` above since this is changing global state, but global state should only change when `pyhf.set_backend(...)` is called. Maybe we can discuss in a separate issue.

Reply:
I don't understand this to be honest. Can you not use `self.astensor(_par, dtype='float')` ?

Reply:
unfortunately, it changes when the torch backend is **initialized**. In practice one probably doesn't create multiple backend objects, but it's done for the tests
https://github.com/scikit-hep/pyhf/blob/64dbce070156945f6bb524f79ee6e23d26b52f01/tests/conftest.py#L79-L89
So when a test runs, the global state is set to 64b precision. So also the 32b pytroch test will run with 64b. Even weirder, if it happens that another test that creates a pytorch backend, e.g. [`test_set_tensor_precision`](https://github.com/scikit-hep/pyhf/blob/master/tests/test_tensor.py#L387) runs before then the tests for both pytorch backends will be both 32b (since that was the last one tested there).
So i thought the easiest solution would be not to rely on such a global setting. The alternative would be to build something in that ensures this call is made once the backend is **changed**.

Reply:
```python
>>> import torch
>>> torch.set_default_dtype(torch.float64)
>>> torch.as_tensor([1.0])
tensor([1.])
>>> torch.as_tensor([1.0]).dtype
torch.float64
>>> torch.set_default_dtype(torch.float32)
>>> torch.as_tensor([1.0]).dtype
torch.float32
```

it seems to be switching for me when the dtype changes for the backend.

```python
>>> import pyhf
>>> pyhf.set_backend(pyhf.tensor.pytorch_backend(precision='32b'))
>>> pyhf.tensorlib.astensor([1.]).dtype
torch.float32
>>> pyhf.set_backend(pyhf.tensor.pytorch_backend(precision='64b'))
>>> pyhf.tensorlib.astensor([1.]).dtype
torch.float64
```

Can you clarify?

Reply:
The problem occurs when you create the backend objects both before you do the first call to `set_backend`

```pycon
>>> import pyhf
>>> backend64 = pyhf.tensor.pytorch_backend(precision="64b")
>>> backend32 = pyhf.tensor.pytorch_backend(precision="32b")
>>> pyhf.set_backend(backend64)
>>> pyhf.tensorlib.astensor([1.]).dtype
torch.float32
```

Reply:
Ok, I understand now. Can you file an issue with that minimal code example? I know how to fix it.

Reply:
see #921

Reply:
@nikoladze Can you rebase this off of (for you) `upstream/master`? I think that given the resolution in Issue #921 this should be resolved.

Reply:
Since the resolution in #922 i merged `upstream/master` and changed back to the simpler version that relies on the pytorch default dtype and should work now.